### PR TITLE
Enable testing against JDK 13 EA builds

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds JDK 13 to the CI rotation for testing. For now, we will be testing against JDK 13 EA builds.

### CODE REVIEW

**Review by alpar-t — APPROVED** [review]
LGTM 

Just a suggestion: 
It might be worth doing `s/openjdk12/openjdk13` in `.ci/java-versions.properties` and reverting it before merge, to get a CI run with the new version as part of the PR checks. 
I'm fine if you did this locally too. We had these jobs fail for a while with trivial failures for a number of days until everything was sorted out. Granted I do expect this will become progressively better with the accelerated java releases so just a suggestion.

**Comment by jasontedor** [comment]
@atorok It was good a couple of weeks ago, yet I just tried on the latest build (14) just published and there are some issues. I will update this PR.

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
@atorok It looks like Gradle is trying to run on JDK 13 which it is not ready for yet. Is it possible in our infrastructure to have it be that Gradle runs on JDK 12 but we set `compiler.home` to 13? I suppose we need another variable in the .ci variables and support for respecting that in CI? Otherwise, this change can not go in yet and we can not start testing against JDK 13.

**Comment by jasontedor** [comment]
Relates gradle/gradle#8681

**Comment by atorok** [comment]
@jasontedor there isn't currently a way to do that and I was really hopping we could keep it that way as I think this adds a lot of confusion to the CI setup as to which version of Java  is used where and why this is necessary - most of the time on the surface it seems superfluous since the compiler and build java will be the same. 

Do we really think we need this much testing of the compiler this early on ?
I'm asking because if can't build with openjdk13 nobody else can :) so it would seem that 
building with openjdk12 and setting  runtime java to openjdk13 would be sufficient for the time being. 
Of course it would be a nice courtesy to the JDK team to do it, but we could do it as soon as the tool chains we use support it without these workarounds, especially that with the increased release cycles these tools will probably be forced to accelerate their releases as well.

**Comment by jasontedor** [comment]
JDK 13 will reach ramp down phase already two months from now. I do not want to be in a position where we are at the last minute investigating any issues we might have compiling with JDK 13, solely because we were blocked on upstream when we could have worked around it.

**Comment by atorok** [comment]
I think the best path forward is to test with runtime java on the PR like you have now, revert that before merge, and then set up CI to use different JVMs for Gradle and the compiler, but do it only specifically for Java 13 so it's clear that this is a special case we have to cover now instead of something we want to support on the on going bases. 
This will be based on the premise that upstream will get better at it and the problem will go  away. How does that sound ?

**Comment by atorok** [comment]
I tested on this branch to make sure that compiling using a different jdk version than running Gradle still works: 
`./gradlew -Dcompiler.java=13 precommit`
```
=======================================
Elasticsearch Build Hamster says Hello!
  Gradle Version        : 5.3
  OS Info               : Linux 5.0.6-arch1-1-ARCH (amd64)
  Compiler JDK Version  : 13 (Oracle Corporation 13-ea [OpenJDK 64-Bit Server VM 13-ea+15])
  Compiler java.home    : /home/alpar/opt/openjdk-13
  Runtime JDK Version   : 13 (Oracle Corporation 13-ea [OpenJDK 64-Bit Server VM 13-ea+15])
  Runtime java.home     : /home/alpar/opt/openjdk-13
  Gradle JDK Version    : 12 (Oracle Corporation 12 [OpenJDK 64-Bit Server VM 12+33])
  Gradle java.home      : /home/alpar/opt/jdk-12
  Random Testing Seed   : CD6E33CF128AD072
=======================================
```

**Comment by jasontedor** [comment]
Thank you @atorok.

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/default-distro

File: .ci/java-versions.properties
Comment (jasontedor) [inline .ci/java-versions.properties]: Note that this is temporary for testing this pull request. It will be removed before merging.

File: .ci/java-versions.properties
Comment (jasontedor) [inline .ci/java-versions.properties]: Note that this is temporary for testing this pull request. It will be removed before merging.

File: buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java
Comment (jasontedor) [inline buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java]: Note that this is temporary for testing this pull request. It will be removed before merging.

File: buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java
Comment (jasontedor) [inline buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java]: Note that this is temporary for testing this pull request. It will be removed before merging.

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Pull request merged**
